src/cudamatrix/cu-matrix-lib.h

```diff
@@ -26,6 +26,7 @@
 #include "cudamatrix/cu-matrix.h"
 #include "cudamatrix/cu-sp-matrix.h"
 #include "cudamatrix/cu-tp-matrix.h"
+#include "cudamatrix/cu-sparse-matrix.h"
```

Comment:
I think some compilers report undefined symbol when seeing a reference to `var` in a `template calss <B> class D : B { ... }` when `D` does not declare `var`. I have no idea if this is a consequence of the C++ standard, and, if it is, an intended one. The new C++ `concept` TR WIP makes templates categorial, I think (C++ meets Haskell, for better or worse).

I did not ever check the public version of Google standards, but apparently it does require the alphabetical sorting, see[Include order](https://google-styleguide.googlecode.com/svn/trunk/cppguide.html#Names_and_Order_of_Includes). It would be a very good idea to enforce even if it did not, as duplicate includes are almost invisible otherwise.